File: frontend/src/microservices/content/objects/ExploreSearchListPlannerResponse.ts
// ExploreSearchListPlannerResponse：content 域探索搜索列表返回对象。

import type { ExploreSearchResultPlannerResponse } from './ExploreSearchResultPlannerResponse'
export type ExploreSearchListPlannerResponse = {
  results: ExploreSearchResultPlannerResponse[]
}
export const exploreSearchListPlannerResponseFromJson = (json: string): ExploreSearchListPlannerResponse =>
  JSON.parse(json) as ExploreSearchListPlannerResponse
export const exploreSearchListPlannerResponseToJson = (value: ExploreSearchListPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/ExploreSearchListPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ExploreSearchListPlannerResponse(results: List[ExploreSearchResultPlannerResponse])
object ExploreSearchListPlannerResponse:
  given sourceEncoder: Encoder[ExploreSearchListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ExploreSearchListPlannerResponse] = deriveDecoder
